src/compiler/tsbuild.ts

```diff
@@ -1058,7 +1063,7 @@ namespace ts {
             }
 
             const inputTime = host.getModifiedTime(inputFile);
-            if (inputTime > newestInputFileTime) {
+            if (inputTime !== undefined && inputTime > newestInputFileTime) {

```

Comment:
Can you do a single check up-front? Plus, it doesn't seem like we're now actually handling `undefined`, we're just doing observably nothing.

Reply:
Ok, will move.

Reply:
Ok, will move. Maybe would be better to set a default value in order to avoid checking on undefined, like it was done in this case https://github.com/a-tarasyuk/TypeScript/blob/550710d2a2ce019dfc7ba8120b112e635fb5d9dc/src/compiler/sys.ts#L306. what do you think?

Reply:
This condition was changed because program uses https://github.com/a-tarasyuk/TypeScript/blob/550710d2a2ce019dfc7ba8120b112e635fb5d9dc/src/compiler/program.ts#L194 `sys.getModifiedTime` where was added `undefined`

Reply:
If we haven't run into `undefined` errors before, we're likely always be passing valid paths. So dropping every exception at the implementation of `function getModifiedTime` using `catch (e) { return undefined; }` was probably the wrong choice in the first place.

Reply:
This condition has been changed because the program uses https://github.com/a-tarasyuk/TypeScript/blob/550710d2a2ce019dfc7ba8120b112e635fb5d9dc/src/compiler/program.ts#L194 `sys.getModifiedTime` where has been added `undefined`

Reply:
@andy-ms Thanks. 

> So dropping every exception at the implementation of `function getModifiedTime` using `catch (e) { return undefined; }` was probably the wrong choice in the first place.

Do I need to remove this `catch`? If yes, how to handle incorrect file modified time? What do you think which the best option for this case?

Reply:
@sheetalkamat Thoughts?

Reply:
@sheetalkamat could you check these changes? thanks in advance.

Reply:
@andy-ms we shouldnt assume that getModifiedTime would be called only on present files.
@a-tarasyuk You are on right track. Please use ```inputTime = host.getModifiedTime(inputFile) || missingFileModifiedTime;``` instead of additional checks.

Reply:
@sheetalkamat thank you very much. I'll add changes and re-open PR.